docs/kb/semgrep-supply-chain/sss-python-lockfiles.md

```diff
@@ -4,48 +4,58 @@ tags:
   - Semgrep Supply Chain
   - Python
   - Knowledge Base
-  - Lock files
+  - Lockfiles
   - requirements.txt
   - Pipfile.lock
   - Poetry.lock
 ---
 
-# Generating Python lock files for Semgrep Supply Chain scans
+# Generating Python lockfiles for Semgrep Supply Chain scans
 
-Python has different mechanisms to detect all packages used in a project. 
-
-Semgrep Supply Chain can use any of the following methods to correctly parse your dependencies and detect vulnerabilities:
+To correctly scan all dependencies in a project, Semgrep Supply Chain requires a Python lockfile. This article describes methods to generate the following Python lockfiles:
 
 * `requirements.txt`
 * `Pipfile.lock`
-* Poetry
-You can use these three ways (requirements.txt, Pipfile.lock, Poetry) to get a successful semgrep supply-chain scan.
+* `Poetry .lock`
+
+You can use any of these three lockfiles to get a successful Semgrep Supply Chain scan.
+
+## Generating `requirements.txt`
```

Comment:
Since there are several ways of generating `requirements.txt` I created a general "Generating requirements.txt" then put them under that heading